lua/lazyvim/plugins/extras/lang/rust.lua

```diff
@@ -45,82 +45,58 @@ return {
   },
 
   {
-    "simrat39/rust-tools.nvim",
-    lazy = true,
-    opts = function()
-      local ok, mason_registry = pcall(require, "mason-registry")
-      local adapter ---@type any
-      if ok then
-        -- rust tools configuration for debugging support
-        local codelldb = mason_registry.get_package("codelldb")
-        local extension_path = codelldb:get_install_path() .. "/extension/"
-        local codelldb_path = extension_path .. "adapter/codelldb"
-        local liblldb_path = ""
-        if vim.loop.os_uname().sysname:find("Windows") then
-          liblldb_path = extension_path .. "lldb\\bin\\liblldb.dll"
-        elseif vim.fn.has("mac") == 1 then
-          liblldb_path = extension_path .. "lldb/lib/liblldb.dylib"
-        else
-          liblldb_path = extension_path .. "lldb/lib/liblldb.so"
-        end
-        adapter = require("rust-tools.dap").get_codelldb_adapter(codelldb_path, liblldb_path)
-      end
-      return {
-        dap = {
-          adapter = adapter,
-        },
-        tools = {
-          on_initialized = function()
-            vim.cmd([[
-                  augroup RustLSP
-                    autocmd CursorHold                      *.rs silent! lua vim.lsp.buf.document_highlight()
-                    autocmd CursorMoved,InsertEnter         *.rs silent! lua vim.lsp.buf.clear_references()
-                    autocmd BufEnter,CursorHold,InsertLeave *.rs silent! lua vim.lsp.codelens.refresh()
-                  augroup END
-                ]])
-          end,
+    "mrcjkb/rustaceanvim",
+    version = "^3", -- Recommended
+    ft = { "rust" },
+    opts = {
+      server = {
+        on_attach = function(client, bufnr)
+          -- register which-key mappings
```

Comment:
Could you add: `vim.lsp.inlay_hint.enable(bufnr, true)` to have inlay hints enable by default, like with `rust-analyzer`. 

This will only work with neovim's builtin inlay hints, which I think are only in nightly. That's why @trevarj was saying in the discussion "it is probably not worth migrating until nvim 0.10".

Reply:
shouldn't that be part of core LazyVim LspAttach functionality when it's running with neovim >= 0.10.x? see here: https://github.com/LazyVim/LazyVim/blob/879e29504d43e9f178d967ecc34d482f902e5a91/lua/lazyvim/plugins/lsp/init.lua#L119-L121

Reply:
Are they being enabled by default for you? I tried late last night, and they weren't for me. Maybe I had other issues, I'll try again.

Reply:
hmmm might be related to this: https://github.com/appelgriebsch/Nv/blob/main/lua/plugins/lang.lua#L6

Reply:
When I call `:lua print(vim.fn.has('nvim-0.10'))` it prints `1`

```
λ ➜ nvim --version
NVIM v0.10.0-dev-1890+g1cf51a07a
Build type: RelWithDebInfo
LuaJIT 2.1.1702233742
Run "nvim -V1 -v" for more info
```

`:lua print(vim.fn.has('v0.10.0-dev-1890+g1cf51a07a'))` returns `0`

Reply:
They only enable for me when I start typing in insert mode and then <esc>ing and saving

Reply:
For neovim < 0.10, there's [a separate plugin](https://github.com/lvimuser/lsp-inlayhints.nvim) that enables inlay hints.
There's also [one by simrat39](https://github.com/simrat39/inlay-hints.nvim), the rust-tools author, which is based on the implementation in rust-tools.

The existence of those two plugins + the built-in support after nvim 0.10 was the reason I decided to drop inlay hints from rustaceanvim.

I don't remember how good the performance was in rust-tools, but compared to the nvim 0.10 implementation, it [appears to be quite slow with both plugins](https://github.com/mrcjkb/rustaceanvim/discussions/46#discussioncomment-7697717).

I suppose you could in theory add one of the plugins as a plugin spec with `optional = vim.fn.has('nvim-0.10')`.